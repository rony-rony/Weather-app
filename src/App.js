import React from "react";

import Titles from "./components/Titles";
import Form from "./components/Form";
import Weather from "./components/Weather";

const API_KEY = "d2430ee85ee2055d290d0e63e943f7e3";

class App extends React.Component {
  state = {
    temperature: null,
    city: null,
    country: null,
    humidity: null,
    description: null,
    error: null
  }
   

  getWeather = async (e) => {
    e.preventDefault();
    const valueOfCityAndCountry = {
      city: e.target.elements.city.value,
      country: e.target.elements.country.value
    };
    const city = valueOfCityAndCountry.city;
    const country = valueOfCityAndCountry.country;
    const api_call = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city},${country}&appid=${API_KEY}`);

    const data = await api_call.json();

    var clearFields = () => {
      this.setState({
        temperature: null,
        city: null,
        country: null,
        humidity: null,
        description: null,
        error: "Please enter the values."
      });
    }

    if (city && country) {
      this.setState({
        temperature: data.main.temp,
        city: data.name,
        country: data.sys.country,
        humidity: data.main.humidity,
        description: data.weather[0].description,
        error: ""
      });
    } else {
      clearFields();
    }
  }
  render() {
    return (
      <React.Fragment>
        <div className="wrapper">
          <div className="main">
            <div className="container">
              <div className="row">
                <div className="col-xs-5 title-container">
                  <Titles />
                </div>
                <div className="col-xs-7 form-container">
                  <Form getWeather={this.getWeather} />
                  <Weather
                    temperature={this.state.temperature}
                    city={this.state.city}
                    country={this.state.country}
                    humidity={this.state.humidity}
                    description={this.state.description}
                    error={this.state.error}
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        </React.Fragment>

    );
  }
};

export default App;
