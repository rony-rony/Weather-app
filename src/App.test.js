import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';
import Weather from './components/Weather';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should render a Weather', () => {
  const wrapper = shallow(
      <Weather>Hello Jest!</Weather>
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders App wrapper without crashing', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
