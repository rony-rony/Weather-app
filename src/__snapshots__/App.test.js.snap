// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App wrapper without crashing 1`] = `ShallowWrapper {}`;

exports[`should render a Weather 1`] = `ShallowWrapper {}`;
